src/Altinn.App.Api/Controllers/Conventions/EnumAsNumberFormatter.cs

```diff
@@ -0,0 +1,71 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Altinn.App.Api.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Altinn.App.Api.Controllers.Conventions;
+
+internal class EnumAsNumberFormatter : SystemTextJsonOutputFormatter
+{
+    internal EnumAsNumberFormatter(string settingsName, JsonOptions options)
+        : base(CreateSerializerOptions(options))
+    {
+        SettingsName = settingsName;
+    }
+
+    internal string SettingsName { get; }
+
+    public override bool CanWriteResult(OutputFormatterCanWriteContext context)
+    {
+        if (context.HttpContext.GetJsonSettingsName() != SettingsName)
+        {
+            return false;
+        }
+
+        return base.CanWriteResult(context);
+    }
+
+    private static JsonSerializerOptions CreateSerializerOptions(JsonOptions options)
+    {
+        var newOptions = new JsonSerializerOptions(options.JsonSerializerOptions);
+        newOptions.Converters.Add(new EnumToNumberJsonConverterFactory());
+        return newOptions;
+    }
+}
+
+internal class EnumToNumberJsonConverterFactory : JsonConverterFactory
+{
+    public override bool CanConvert(Type typeToConvert)
+    {
+        return typeToConvert.IsEnum;
+    }
+
+    public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+    {
+        var instance =
+            Activator.CreateInstance(typeof(EnumToNumberJsonConverter<>).MakeGenericType(typeToConvert))
+            ?? throw new InvalidOperationException($"Failed to create converter for type {typeToConvert.FullName}");
+        return (JsonConverter)instance;
+    }
+}
+
+internal class EnumToNumberJsonConverter<TEnum> : JsonConverter<TEnum>
```

Comment:
Do we need a custom converter? Isn't number encoding the default?

Reply:
If I understand correctly, the issue is that service owners might change the default, affecting our apis.

Reply:
Oh! you are thinking about using [`JsonNumberEnumConverter<TEnum>`](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonnumberenumconverter-1?view=net-8.0) instead?

Reply:
Sure. Its generic clause is struct instead of struct, Enum.